src/HalfIntegers.jl

```diff
@@ -561,6 +562,38 @@ ishalfinteger(::HalfIntegerOrInteger) = true
 ishalfinteger(::AbstractIrrational) = false
 ishalfinteger(::Missing) = missing
 
+"""
+    ishalfoddinteger(x)
+
+Test whether `x` is numerically equal to some half-odd-integer.
+
+# Examples
+
+```jldoctest
+julia> ishalfoddinteger(3.5)
+true
+
+julia> ishalfoddinteger(2)
+false
+
+julia> ishalfoddinteger(1//3)
+false
+```
+"""
+ishalfoddinteger(x) = _isodd(twice(x))
+ishalfoddinteger(x::Rational) = denominator(x) == 2
+ishalfoddinteger(::Integer) = false
+ishalfoddinteger(::AbstractIrrational) = false
+ishalfoddinteger(::Missing) = missing
+
+if VERSION ≥ v"1.7"
```

Comment:
I think we don't need `@static` macro here because this is outside of a function.